symengine/tests/cwrapper/test_cwrapper.c

```diff
@@ -238,6 +238,23 @@ void test_get_type() {
     basic_free(y);
 }
 
+void test_hash() {
+    basic x1, x2, y;
+    basic_init(x1);
+    basic_init(x2);
+    basic_init(y);
+    symbol_set(x1, "x");
+    symbol_set(x2, "x");
+    symbol_set(y, "y");
+
+    SYMENGINE_C_ASSERT(basic_hash(x1) == basic_hash(x2));
+    if (basic_hash(x1) != basic_hash(y)) SYMENGINE_C_ASSERT(basic_neq(x1,y));
```

Comment:
Right. In practice, the hashes will actually be always different, if the hash function is good. It's not a very good test, but it was written in a way to represent this logic.

A better test would be to figure out how to use some intentionally bad hash function, like returning always "1" for all objects, and then test the machinery, i.e. making sure that `basic_eq` still returns False if the objects are not equal (even though they have the same hash). I don't know how to do change the hash function in the test only though.